frontend/appflowy_flutter/lib/plugins/database/application/cell/bloc/date_cell_editor_bloc.dart

```diff
@@ -115,241 +112,129 @@ class DateCellEditorBloc
-                  .add(ReminderEvent.remove(reminderId: state.reminderId!));
-              await _updateDateData(reminderId: "");
-              emit(state.copyWith(reminderOption: option));
-            } else if (state.dateTime != null &&
-                (state.reminderId?.isNotEmpty ?? false)) {
-              final scheduledAt = option.fromDate(state.dateTime!);
-
+            } else if (state.dateTime != null) {
               // Update reminder
+              final scheduledAt =
+                  option.getNotificationDateTime(state.dateTime!);
               _reminderBloc.add(
                 ReminderEvent.update(
                   ReminderUpdate(
-                    id: state.reminderId!,
+                    id: state.reminderId,
                     scheduledAt: scheduledAt,
                     includeTime: true,
                   ),
                 ),
               );
             }
           },
-          // Empty String signifies no reminder
-          removeReminder: () async => _updateDateData(reminderId: ""),
+          removeReminder: () async {
+            await _updateDateData(reminderId: "");
+          },
         );
       },
     );
   }
 
   Future<void> _updateDateData({
     DateTime? date,
-    String? timeStr,
     DateTime? endDate,
-    String? endTimeStr,
-    bool? includeTime,
-    bool? isRange,
     String? reminderId,
   }) async {
-    // make sure that not both date and time are updated at the same time
-    assert(
-      !(date != null && timeStr != null) ||
-          !(endDate != null && endTimeStr != null),
+    final result = await _dateCellBackendService.update(
+      date: date,
+      endDate: endDate,
+      reminderId: reminderId,
     );
+    result.fold((_) {}, (err) => Log.error(err));
+  }
 
-    // if not updating the time, use the old time in the state
-    final String? newTime = timeStr ?? state.timeStr;
-    final DateTime? newDate = timeStr != null && timeStr.isNotEmpty
-        ? state.dateTime ?? DateTime.now()
-        : _utcToLocalAndAddCurrentTime(date);
-
-    // if not updating the time, use the old time in the state
-    final String? newEndTime = endTimeStr ?? state.endTimeStr;
-    final DateTime? newEndDate = endTimeStr != null && endTimeStr.isNotEmpty
-        ? state.endDateTime ?? DateTime.now()
-        : _utcToLocalAndAddCurrentTime(endDate);
+  Future<void> _updateIsRange(bool isRange) async {
+    final dateTime = state.dateTime == null ? DateTime.now().withoutTime : null;
+    final endDateTime = dateTime;
 
     final result = await _dateCellBackendService.update(
-      date: newDate,
-      time: newTime,
-      endDate: newEndDate,
-      endTime: newEndTime,
-      includeTime: includeTime ?? state.includeTime,
-      isRange: isRange ?? state.isRange,
-      reminderId: reminderId ?? state.reminderId,
+      date: dateTime,
+      endDate: endDateTime,
+      isRange: isRange,
     );
+    result.fold((_) {}, (err) => Log.error(err));
```

Comment:
If we don't need to handle the success case, we can use `result.onFailure` instead.

Reply:
Done